homework-submission/week5-homework/src/stores/TodoStore.js

```diff
@@ -0,0 +1,148 @@
+import { configure, observable, action, computed, runInAction } from "mobx";
+import uuidv4 from "uuid/v4";
+const url = "https://hyf-react-api.herokuapp.com";
+
+configure({ enforceActions: "observed" });
+class TodoClass {
+  @observable
+  todos = [];
+  @observable
+  state = "Loading";
+
+  @action
+  async getTodos() {
+    this.state = "Loading";
+    this.todos = [];
+    try {
+      let todos = await this.getData();
+      runInAction(() => {
+        this.state = "Done";
+        this.todos = todos;
+      });
+    } catch (error) {
+      runInAction(() => {
+        this.state = "error";
+      });
+    }
+  }
+
+  @action
+  toggleDone = async id => {
+    let todoItem = this.todos.find(todo => todo._id === id);
+    try {
+      await this.handleFetch(`${url}/todos/${id}`, "PATCH", {
```

Comment:
if you will assign this line to a variable we may have a look to the next comment...